app/assets/javascripts/miq_explorer.js

```diff
@@ -0,0 +1,185 @@
+ManageIQ.explorer = {}
+
+ManageIQ.explorer.updateElement = function(element, options) {
+  if ( _.isString(options.legend) ) {
+    $('#' + element).html(options.legend);
+  } else if ( _.isString(options.title) ) {
+    $('#' + element).attr( {'title': options.title});
+  }
+};
+
+ManageIQ.explorer.buildCalendar = function(options) {
+  var skip_days = _.isObject(options.skipDays) ? options.skipDays : undefined;
+
+  ManageIQ.calendar.calDateFrom = options.dateFrom; //# FIXME js_format_date
+  ManageIQ.calendar.calDateTo   = options.dateTo;
+  ManageIQ.calendar.calSkipDays = skip_days;
+
+  miqBuildCalendar();
+}
+
+ManageIQ.explorer.lock_tree = function(tree, lock) {
+  $('#' + tree + 'box').dynatree(lock ? 'disable' : 'enable');
+  miqDimDiv('#' + tree + '_div', lock);
+}
+
+ManageIQ.explorer.clear_search_show_or_hide = function(show) {
```

Comment:
@martinpovolny isn't there a better name for this? `clear_search_toggle` for example